js/liquid.js

```diff
@@ -105,6 +105,339 @@ module.exports = class liquid extends Exchange {
     async fetchMarkets () {
         let markets = await this.publicGetProducts ();
         let result = [];
+        // see: https://quoine.zendesk.com/hc/en-us/articles/360004828252-Minimum-withdrawal-amount-Minimum-order-quantity
+        let minOrderAmounts = {
+            'BTC': 0.001,
+            'ETH': 0.01,
+            'BCH': 0.01,
+            'DASH': 0.01,
+            'QTUM': 0.1,
+            'NEO': 0.1,
+            'UBTC': 0.01,
+            'XRP': 1,
+            'RKT': 1,
+            'QASH': 1,
+        };
+        let defaultPrecisions = { 'price': 8, 'amount': 8 };
+        let precisions = {
+            '1WO/BTC': defaultPrecisions,
+            '1WO/ETH': defaultPrecisions,
+            '1WO/QASH': defaultPrecisions,
+            'ADH/BTC': defaultPrecisions,
+            'ADH/ETH': defaultPrecisions,
+            'ADH/QASH': defaultPrecisions,
+            'ALX/BTC': defaultPrecisions,
+            'ALX/ETH': defaultPrecisions,
```

Comment:
Ouch...
This duplication looks nasty...
This is how we do this type of thing to avoid hardcoding: https://github.com/ccxt/ccxt/blob/master/js/kraken.js#L216
I'll have to rewrite this as we've deprecated hardcoding these values.

Reply:
Agreed, but I don't know where to get the precisions values from other than scaling the order books.  I don't think Liquid publishes them anywhere.  The min order amounts are on a webpage - I've included a comment to the page I used.

Reply:
Yep, check that link to kraken above, it shows how we fetch their zendesk html, break it into pieces and parse the values from it. This is similar to what we are going to do for liquid.

Reply:
But anyways, no worries, thx for helping us with this! I hope to rewrite the parsing and will merge this in a day or two.

Reply:
Ok.  That will presumably work for the min order amounts.  But what about the precisions?

Reply:
> But what about the precisions?

How did you fill those values for precisions? Tried create_order with zeroes?

Reply:
By scraping the order books

Reply:
@npomfret ok, got it, we will check for that along the way as well... If we don't find an endpoint for that, then, I guess, we're only left with hardcoding. However, those values have to be checked with `create_order` first to see if it tolerates them and find incorrect ones. In other words, deriving values from orderbooks can sometimes give false results, depending on the orderbook depth.

Reply:
I made the (possibly incorrect) assumption that a price in the order book would not be too precise.  I can see that they might not be precise enough though.

FYI - Liquid appears to accept overly precise prices - they get rounded on the exchange.